app/src/androidTest/java/com/urbandroid/sleep/addon/stats/model/socialjetlag/SocialJetlagStatsTest.java

```diff
@@ -71,4 +76,169 @@ public void testGetRecordIrregularity() {
 
         assertThat(stats.getRecordIrregularity(record)).isEqualTo( ((6.625f - 3.5f) + (8.5f - 6.5f)) / 2f );
     }
+
+    @Test
+    public void calculateSRIEdgeCases() {
+        SocialJetlagStats  stats = new SocialJetlagStats(cr, false);
+
+        // one bitset empty
+        BitSet day1Interval = new BitSet(1440);
+        BitSet day2Interval = new BitSet();
+        day1Interval.set(0,720);
+        Assert.assertEquals(1.0, stats.calculateSRI(day1Interval, day2Interval), .0001);
+
+        // awake 24 hours
+        BitSet interval3 = new BitSet(1440);
+        BitSet interval4 = new BitSet(1440);
+        interval3.set(0,1440);
+        Assert.assertEquals(0, stats.calculateSRI(interval3, interval4), .0001);
+
+        // awake 0 hours
+        BitSet interval5 = new BitSet(1440);
+        BitSet interval6 = new BitSet(1440);
+        Assert.assertEquals(1.0, stats.calculateSRI(interval5, interval6), .0001);
+    }
+
+    @Test
+    public void testCalculateSRI() {
+        BitSet day1Interval = new BitSet(1440);
+        BitSet day2Interval = new BitSet(1440);
+        day1Interval.set(0,720);
+        day2Interval.set(0,720);
+        // testing for when the intevals are the same
```

Comment:
maybe add : testing for same intervals in day 1 and day 2  to emphasize pairwise date comparisons